packages/utils/src/envelope.ts

```diff
@@ -36,24 +36,73 @@ export function forEachEnvelopeItem<E extends Envelope>(
   });
 }
 
+// Combination of global TextEncoder and Node require('util').TextEncoder
+interface TextEncoderInternal extends TextEncoderCommon {
+  encode(input?: string): Uint8Array;
+}
+
 /**
- * Serializes an envelope into a string.
+ * Serializes an envelope.
  */
-export function serializeEnvelope(envelope: Envelope): string {
-  const [headers, items] = envelope;
-  const serializedHeaders = JSON.stringify(headers);
-
-  // Have to cast items to any here since Envelope is a union type
-  // Fixed in Typescript 4.2
-  // TODO: Remove any[] cast when we upgrade to TS 4.2
-  // https://github.com/microsoft/TypeScript/issues/36390
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  return (items as any[]).reduce((acc, item: typeof items[number]) => {
-    const [itemHeaders, payload] = item;
-    // We do not serialize payloads that are primitives
-    const serializedPayload = isPrimitive(payload) ? String(payload) : JSON.stringify(payload);
-    return `${acc}\n${JSON.stringify(itemHeaders)}\n${serializedPayload}`;
-  }, serializedHeaders);
+export function serializeEnvelope(envelope: Envelope, textEncoder?: TextEncoderInternal): string | Uint8Array {
+  const utf8 = textEncoder || new TextEncoder();
```

Comment:
`TextEncoder` not supported in ie11 :/ 

Could we make a small polyfill or something?

Reply:
This looks well used but would add ~500 bytes to the browser bundle. 
https://github.com/LinusU/encode-utf8/blob/master/index.js

Is there a way that we can only include this for the es5 bundle?

Reply:
There's [this](https://www.npmjs.com/package/text-encoding) proper polyfill for the full `TextEncoder/TextDecoder` API but it's 18kB.

Could this just be added to the list of polyfills required for IE11 here?
https://docs.sentry.io/platforms/javascript/troubleshooting/supported-browsers/#support-for--ie-11

Reply:
>  Remembered that's going soon.

We will still emit an es5 bundle though. @lobsterkatie any ideas on how to best proceed here?

Reply:
I think we add it to the list in docs (we'll need to change the wording a bit, to say that we again don't polyfill anything and people need to do it for themselves) and otherwise don't worry about it. Anyone still trying to support IE11 is by necessity already deep in Polyfill Land, and the 99% of other people will benefit from us not adding it to the bundle size.

Reply:
I chatted to Abhi about this earlier and realised that `TextEncoder` is only actually required if you add an attachment. Without an attachment, envelopes are still created as a string.

So if you want attachments to work on ie11 you'll need to polyfill `TextEncoder`.

I do need to add a test to ensure basic event/transaction envelopes do not hit that path.